.github/workflows/build_and_test.yml

```diff
@@ -182,25 +199,6 @@ jobs:
       run: ./scripts/shmem_limits_macos.sh
     - name: Run Tests
       run: cargo test
-  macos-fuzzers:
-    runs-on: macOS-latest
-    steps:
-    - uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        toolchain: stable
-    - uses: Swatinem/rust-cache@v1
```

Comment:
I think the rust-cache could work too if we set the paths up correctly (every fuzzer needs caching)

Reply:
IMO rust-cache is used to cache the download directory(like `~/.cargo/bin`) in GitHub Action so that a new workflow doesn't need to download the same crate again. But currently, our problem is to share the compilation results between different cargo projects **in a single workflow**, so we may need to change the cache mechanism of `rust` other than `github action`.

Reply:
This blog may help 
https://www.infinyon.com/blog/2021/04/github-actions-best-practices/#using-the-github-workflows-build-matrix

Reply:
`sccache` didn't achieve the desired performance boost. So now I take every example in fuzzer as a single job(that's exactly the multi-thread mentioned in #565 )
I notice that the thread number is limited. Is it suitable to merge the Job `windows`, `ubuntu` and `Linux` task into `common`? It's feasible by setting the `if` condition in steps.

Reply:
I don't think we have unlimited runs, only about 14 max.
Caching the download/build artifacts for every single fuzzer should be the best solution IMHO

Reply:
I have had the same concern before, but GitHub action only limits the max concurrent and single run time.  The others will be queued until another runner is free.
here is the doc:
https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration

Reply:
I have also enabled Action in my fork and it works
https://github.com/syheliel/LibAFL/runs/5701259719?check_suite_focus=true

Reply:
but if they get queued it doesn't help us much, the overall time will remain the same

Reply:
Because you have more than one runner 😂 
In my [run](https://github.com/syheliel/LibAFL/runs/5701259719?check_suite_focus=true), the job `ubuntu` is now the most time-consuming job(about 1.9h)
But in PR #578 ,one complete `macos-fuzzer` job will take 3.5h

Reply:
Now I understand...  My original idea is to split examples into pieces so that we can take advantage of all the runners, but the runners are shared between all PRs. So if one PR occupies all runners, it won't help to reduce the total time.

Reply:
One simple way is to convert the relative path into a specific version in `Cargo.toml-[dependenies]` so that the examples will rerun only when they are modified.
Actually, it's a bit weird when the relative path appears in example dir instead of test dir. Because the user should run the example alone without the `LibAFL` git repo.